### DISCUSSION TITLE
새로운 커밋 메시지 컨벤션을 제안합니다.

### DISCUSSION THREAD

**Discussion by jungwoo3490** [Setaday/Setaday-Client#2] [Ideas]
커밋 메시지 뒤에 (#이슈번호) 를 추가하면 커밋 관련 이슈를 쉽게 확인할 수 있어 더 좋은 방식같은데 어떻게 생각하시나요??
실제로 toss에서 사용하는 방식을 차용한 아이디어예요.

<img width="900" alt="image" src="https://github.com/user-attachments/assets/226c5ecb-3a78-473a-ad08-2b12380574a4">

또 이번에 모노레포를 사용하는데, 구조가 복잡하기 때문에 Commit Type 옆에 어느 위치에서 작업했는지를 같이 명시해주면 좋을 것 같아요!!

ex) chore(root): init turbo / feat(desktop): 커스텀 캘린더 구현 / fix(mobile): 화면 깨짐 오류 수정

다들 어떻게 생각하시나요?!

**Comment by Arooming** [comment]
이슈 번호 붙이는게 훨씬 깔끔하고 어떤 이슈 관련 커밋인지 한 눈에 파악돼서 좋네용  !
그리고 모노레포를 사용해서 구조가 복잡해지는만큼 어디에서 온 커밋인지 파악하기 힘들 것 같은데, 정우가 제안해준 방식처럼 위치를 함께 적어주는 것도 
너무 좋습니당 !!~

**Comment by eunbeann** [comment]
정우가 캡쳐해준 이 커밋 메시지 방식은
수동으로 이슈 번호를 붙인 케이스가 아니라 스쿼시 머지 하면서 [자동으로 PR 번호가 붙은 케이스](https://github.com/toss/es-toolkit/commit/cf65b2c601d8cbf20a532f16d44e23a7a2a4702b
) 같은데요!

스쿼시 머지라면 아주 찬성합니다!!

만약 커밋마다 이슈를 하는 방식이라면 매 커밋마다 이슈 번호를 기억하기 너무 번거롭지 않을까요?

커밋 타입 옆에 작업 위치 명시하는 것도 좋지만 이것 역시 한 PR 작업 안에서는 다른 구조 범위까지 넘어가지 않고 작업을 할 것이라고 예상되는데, 매 커밋보다 PR 제목으로 명시하는 것은 어떤가요? 이 경우에도 스쿼시 머지를 통해서 메인 브랜치 커밋으로 남을 것 같습니다!

<img width="547" alt="image" src="https://github.com/user-attachments/assets/dd9626bb-aea7-4b96-8a90-3fcda1e1102d">

**Reply by jungwoo3490** [reply]
스쿼시 머지를 사용해서 그랬던 거군요...!! 스쿼시 머지를 한다면 은빈이가 제안한 대로 PR 제목에만 작업 위치 남기면 될 것 같아 저도 완전 찬성합니다!! 다른 분들 생각은 어떤가요?!
@Arooming @doyn511

**Reply by doyn511** [reply]
다른 플젝에서 스쿼시머지 써보고 있는데 이슈번호 자동으로 붙어서 좋더라고요! 딱히 저희가 뭘 할 일도 없고 머지 할 때 스쿼시 머지로만 선택해서 하면 되는거라 저도 찬성입니다 !! 모노레포이기에 PR 제목에 작업 위치 남기는 것도 찬성합니다!

**Reply by Arooming** [reply]
옹 스쿼시 머지를 사용해본 적이 없어서 몰랐는데, 커밋 관련 리소스가 조금이라도 줄어드는 방향인 것 같다는 생각이 들어요!!
그래서 저두 비니가 제안해준 내용 좋습니당 !!

**Reply by Arooming** [reply]
++ pr 제목에 작업위치 남기는것두 찬성입니당!
